packages/shared/src/lib/auxiliary/wallet-connect/handlers/sign_message.handler.ts

```diff
@@ -34,14 +36,28 @@ export function handleSignMessage(
     }
     const message = Converter.hexToUtf8(hexMessage)
 
-    openPopup({
-        id: PopupId.SignMessage,
-        props: {
-            message,
-            dapp,
-            account,
-            chain,
-            callback: responseCallback,
-        },
-    })
+    const handler: () => void = () =>
+        openPopup({
+            id: PopupId.SignMessage,
+            props: {
+                message,
+                dapp,
+                account,
+                chain,
+                callback: responseCallback,
+            },
+        })
+
+    if (account.index !== get(selectedAccountIndex)) {
```

Comment:
Use store getter instead of store.

```suggestion
    if (account.index !== getSelectedAccountIndex() {
```